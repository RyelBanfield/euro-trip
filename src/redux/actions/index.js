const LOAD_COUNTRIES_INTO_STORE = 'LOAD_COUNTRIES_INTO_STORE';

// eslint-disable-next-line import/prefer-default-export
export const loadCountriesIntoStore = (countries) => ({
  type: LOAD_COUNTRIES_INTO_STORE,
  payload: countries,
});
