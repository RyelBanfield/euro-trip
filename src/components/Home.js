/* eslint-disable react/jsx-one-expression-per-line */
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCountries } from '../API';
import { loadCountriesIntoStore } from '../redux/actions';

const Home = () => {
  const countries = useSelector((state) => state.countries);

  const dispatch = useDispatch();
  useEffect(() => {
    getCountries()
      .then((countries) => {
        dispatch(loadCountriesIntoStore(countries));
      });
  }, []);

  return (
    <main>
      {countries.length > 0 && (
      <>
        <h1 className="countries-header">European countries</h1>
        <div className="countries">
          {countries.map((country) => (
            <div key={country.name} className="country">
              <Link className="country-link" to={`/countries/${country.id}`}>
                <h2 className="country-name">{country.name}</h2>
              </Link>
              <img src={country.flag} alt={country.name} />
              <p className="country-capital">Capital: {country.capital}</p>
              <p className="country-population">Population: {Number(country.population).toLocaleString()}</p>
              <p className="country-language">Language: {country.language}</p>
              <p className="country-demonym">Demonym: {country.demonym}</p>
              <p className="country-currency">Currency: {country.currency}</p>
            </div>
          ))}
        </div>
      </>
      )}
    </main>
  );
};

export default Home;
