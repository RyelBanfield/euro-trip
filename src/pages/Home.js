import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCountries, getFavourites } from '../API';
import { loadCountriesIntoStore, loadFavouritesIntoStore } from '../redux/actions';
import CountriesSlider from '../components/CountriesSlider';
import SearchBar from '../components/SearchBar';

const Home = () => {
  const countries = useSelector((state) => state.countries);

  const dispatch = useDispatch();

  useEffect(() => {
    getCountries()
      .then((countries) => {
        dispatch(loadCountriesIntoStore(countries));
      });

    getFavourites()
      .then((favourites) => {
        dispatch(loadFavouritesIntoStore(favourites));
      });
  }, []);

  return (
    <main className="home fade-in">
      <h1>European countries</h1>
      {countries.length > 0 && (
        <>
          <CountriesSlider countries={countries} />
          <SearchBar countries={countries} />
        </>
      )}
    </main>
  );
};

export default Home;
