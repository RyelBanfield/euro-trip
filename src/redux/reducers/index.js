import { combineReducers } from 'redux';
import userReducer from './user';
import countriesReducer from './countries';
import imagesReducer from './images';
import favouritesReducer from './favourites';

const rootReducer = combineReducers({
  user: userReducer,
  countries: countriesReducer,
  images: imagesReducer,
  favourites: favouritesReducer,
});

export default rootReducer;
