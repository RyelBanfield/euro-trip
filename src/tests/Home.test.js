import { render, screen } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import rootReducer from '../redux/reducers';
import Home from '../pages/Home';

const store = createStore(rootReducer);

it('renders correctly', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

it('renders main', () => {
  render(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const main = screen.getByRole('main');
  expect(main).toBeInTheDocument();
});
