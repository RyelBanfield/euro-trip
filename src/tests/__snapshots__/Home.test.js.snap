// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="home fade-in"
>
  <header>
    <h1>
      Europe is a big place.
    </h1>
    <p>
      Get information about countries you may visit.
    </p>
  </header>
</main>
`;
